src/cozy/wrappers/sharing/sharing.jsx

```diff
@@ -12,12 +13,36 @@ const Sharing = ({
     onShared
 }) => {
   const [showShareModal, setShowShareModal] = useState(false)
+  const [hasRecipientsToBeConfirmed, setHasRecipientsToBeConfirmed] = useState(false)
+
+  const checkRecipientsToBeConfirmed = async () => {
+    const result = await confirmationMethods.getRecipientsToBeConfirmed()
+
+    console.log({result})
+
+    setHasRecipientsToBeConfirmed(result.length > 0)

```

Comment:
What is the gain of having created this `checkRecipientsToBeConfirmed` method? 

Why not just doing: 
```
confirmRecipient: async (user) => {
      await confirmationMethods.confirmRecipient(user)

      const toBeConfirmerd = confirmationMethods.getRecipientsToBeConfirmed()
   setHasRecipientsToBeConfirmed(toBeConfirmerd.length > 0)

      return result
    },
    ```

Having the setState inside the `checkMethod` bothers me a little

Reply:
> What is the gain of having created this checkRecipientsToBeConfirmed method?

It's because I do the same check/set in `confirmRecipient` and in the `useEffect`.

> Having the setState inside the `checkMethod` bothers me a little

What bothers you? The fact that the method is called `check` and does a `set`? Or the fact that this method does the `set` whatever the name?

Reply:
Both. When I read a code, I prefer to have a the "effect" (setState) visible at the first look. "Hiding" them inside a method seems not good to me. This is why I suggested something. 

I can understand that we do the same thing in another method, but actually I prefer to repeat myself instead of having one method doing too much stuff. 

This is a nit, I approved this PR so if you're not agree with that, let's move on ;)

Reply:
I did the change [here](https://github.com/cozy/cozy-pass-web/compare/b7df4891237b505683409078761ba8c5c49a3428..48ff3c33017467159439554d530de1a74a3db80c]

However I still have to create a method in the useEffect to handle the asynchronism

> This is a nit, I approved this PR so if you're not agree with that, let's move on ;)

It was not sure yet of which one to use. This is the kind of small refactoring I struggle between the choice of making a method with a comprehensible name but hiding the implementation, or the reverse (even if here the name was not ideal).